notifications_api_common/viewsets.py

```diff
@@ -162,46 +169,39 @@ def notify(
         # build the content of the notification
         message = self.construct_message(data, instance=instance)
 
+        self.send_notification.delay(status_code, message)
```

Comment:
this still needs to be wrapped in a `transaction.on_commit` because _other_ unexpected exceptions can still cause database errors, causing the transaction to be rolled back. That would result in a notification being published about something that does not exist.

It's also a race condition - if you publish the notification, (in theory) it's possible something acts on it before the DB transaction has committed and the data is not yet visible to the client (whose request runs in a different, isolated DB transaction).